app/src/main/java/woowacourse/shopping/ui/cart/ShoppingCartViewModel.kt

```diff
@@ -0,0 +1,195 @@
+    }
+
+    override fun navigateToOrder() {
+        if (selectedCartItemsCount.value == 0) return
+
+        val productIds = cartItems.value?.filter { it.checked }?.map { it.id } ?: return
+        val orderAmount = selectedCartItemsTotalPrice.value ?: return
+        val ordersCount = selectedCartItemsCount.value ?: return
+        _navigationOrderEvent.setValue(
+            OrderInformation(
+                productIds,
+                orderAmount,
+                ordersCount,
+            ),
+        )
+    }
+
+    override fun delete(cartItemId: Long) {
+        _deletedItemId.setValue(cartItemId)
+    }
+
+    override fun onIncrease(
+        cartItemId: Long,
+        quantity: Int,
+    ) {
+        thread {
+            cartItemRepository.increaseCartItem(cartItemId, quantity)
+            val currentItems = cartItemRepository.loadCartItems()
+            uiHandler.post {
+                updateCartItems(currentItems)
+                updateTotalPrice()
+            }
+        }
+    }
+
+    override fun onDecrease(
+        cartItemId: Long,
+        quantity: Int,
+    ) {
+        thread {
+            cartItemRepository.decreaseCartProduct(cartItemId, quantity)
+            val currentItems = cartItemRepository.loadCartItems()
+            uiHandler.post {
+                updateCartItems(currentItems)
+                updateTotalPrice()
+                updateSelectedCartItemsCount()
+            }
+        }
+    }
+
+    private fun updateCartItems(currentItems: List<CartItem>) {
+        val cartItems = cartItems.value ?: return
+        _cartItems.value =
+            currentItems.map { cartItem ->
+                cartItem.copy(checked = cartItems.first { it.id == cartItem.id }.checked)
+            }
+    }
+
+    override fun selected(cartItemId: Long) {
+        val selectedItem =
+            cartItems.value?.find { it.id == cartItemId } ?: throw IllegalStateException()
+        val changedItem = selectedItem.copy(checked = !selectedItem.checked)
+
+        _cartItems.value =
+            cartItems.value?.map {
+                if (it.id == cartItemId) {
+                    changedItem
+                } else {
+                    it
+                }
+            }
+        updateTotalPrice()
+        updateSelectedCartItemsCount()
+        _isAllSelected.value = cartItems.value?.all { it.checked }
+    }
+
+    private fun updateTotalPrice() {
+        _selectedCartItemsTotalPrice.value =
+            cartItems.value?.filter { it.checked }?.sumOf {
+                it.product.price * it.quantity
+            }
+    }
+
+    override fun selectedAll() {
+        if (isAllSelected.value == true) {
+            updateCartItemsChecked(checked = false)
+            updateTotalPrice()
+            _isAllSelected.value = false
+            return
+        }
+        updateCartItemsChecked(checked = true)
+        updateTotalPrice()
+        updateSelectedCartItemsCount()
+        _isAllSelected.value = true
```

Comment:
거의 유사한 코드 흐름을 가지고 있다면 중복 코드를 줄이면 좋습니다!
추후에 요구사항에 변경이 발생하면, 수정 포인트가 하나로 줄어드니까요 :)
```suggestion
       val oldItemIsAllSelected = isAllSelected.value ?: false

        updateCartItemsChecked(checked = oldItemIsAllSelected.not())
        updateTotalPrice()
        _isAllSelected.value = oldItemIsAllSelected.not()
        if (oldItemIsAllSelected.not()) {
            updateSelectedCartItemsCount()
        }
```

Reply:
중복을 줄여 보려고 시도하다가 방법이 떠오르지 않았었는 데 이렇게 줄여볼 수도 있겠군요!! 감사합니다~ 😊
[aa4fb7a](https://github.com/woowacourse/android-shopping-order/pull/86/commits/aa4fb7a4c66957613d3b8a13cf7faeff71cbafc7)